internal/resource/url.go

```diff
@@ -283,6 +297,72 @@ func (f *Fetcher) FetchFromOEM(u url.URL, dest FetchWriter) ([]byte, error) {
 	return f.copyAndHash(dest, src)
 }
 
+// FetchFromS3 gets data from an S3 bucket as described by u and writes it into
+// dest, returning a hash (if one is set) or an error. It will attempt to
+// acquire IAM credentials from the EC2 metadata service, and if this fails will
+// attempt to fetch the object with anonymous credentials.
+func (f *Fetcher) FetchFromS3(u url.URL, dest FetchWriter) ([]byte, error) {
+	if f.AWSRegion == "" {
+		return nil, ErrAWSRegionNotSet
+	}
+	sess, err := session.NewSession(&aws.Config{
```

Comment:
The session initialized here is always used to create the `s3Manager.Downloader` in `fetchFromS3WithCreds` regardless of platform. Initializing it here instead of in `fetchFromS3WithCreds` also means the session can be reused if a `EC2RoleRequestError` is encountered and the request is reattempted with anonymous credentials. I can move it into `fetchFromS3WithCreds` though if you think that'd look cleaner.